Src/DependencyCollector/Shared/ComponentCorrelation.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Component Correlation configuration.
+    /// </summary>
+    public class ComponentCorrelation
+    {
+        private ICollection<string> excludedDomains = new SanitizedHostList();
+
+        /// <summary>
+        /// Gets or sets a value indicating whether component correlation collection is enabled. That is should there be an attempt to add component correlation related headers to outgoing responses.
+        /// </summary>
+        public bool CollectionEnabled { get; set; } = true;
```

Comment:
In this case it's not collection - it is propagation. Call it `SetCorrelationHttpHeaders` or something like this